app/src/main/java/com/nextcloud/talk/activities/CallActivity.kt

```diff
@@ -1331,6 +1378,30 @@ class CallActivity : CallBaseActivity() {
         }
     }
 
+    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
+        if (binding!!.popupMenu.visibility == View.VISIBLE) {
+            if (event.action == MotionEvent.ACTION_DOWN) {
+                val buttonLocation = IntArray(2)
+                binding!!.popupMenu.getLocationOnScreen(buttonLocation)
+
+                val popupMenuWidth = binding!!.popupMenu.width
+                val popupMenuHeight = binding!!.popupMenu.height
+
+                val buttonLeft = buttonLocation[0]
+                val buttonTop = buttonLocation[1]
+                val buttonRight = buttonLeft + popupMenuWidth
+                val buttonBottom = buttonTop + popupMenuHeight
+
+                val x = event.rawX
+                val y = event.rawY
+                if (x < buttonLeft || x > buttonRight || y < buttonTop || y > buttonBottom) {
+                    binding!!.popupMenu.visibility = View.GONE
```

Comment:
i think analyzing pixels should be avoided just to hide the button. 
Let's try to solve this via focus handling. I guess/hope onFocusChangeListener could do the job.

Reply:
The current implementation automatically hides the button when a click occurs outside of its boundaries, like in ioS. This can be implemented by using Touch Listeners,  I don't think we can use FocusChangeListener to achieve this functionality.

Reply:
Then lets do it the other way round and hide the button when other layouts are touched.

There is already code for

```
        binding!!.gridview.setOnTouchListener { _, me ->
            val action = me.actionMasked
            if (action == MotionEvent.ACTION_DOWN) {
                animateCallControls(true, 0)
            }
            false
        }
        binding!!.conversationRelativeLayout.setOnTouchListener { _, me ->
            val action = me.actionMasked
            if (action == MotionEvent.ACTION_DOWN) {
                animateCallControls(true, 0)
                binding!!.popupMenu.visibility = View.GONE
            }
            false
        }
```

Just add a line to hide the button inside the `action == MotionEvent.ACTION_DOWN` blocks